exporters/metrics/src/test/java/com/google/cloud/opentelemetry/metric/MetricExporterTest.java

```diff
@@ -0,0 +1,143 @@
+        .setMetricKind(MetricKind.CUMULATIVE)
+        .setValueType(MetricDescriptor.ValueType.INT64)
+        .build();
+    TimeInterval expectedTimeInterval = TimeInterval.newBuilder()
+        .setStartTime(
+            Timestamp.newBuilder().setSeconds(aLongPoint.getStartEpochNanos() / NANO_PER_SECOND).setNanos(0).build())
+        .setEndTime(
+            Timestamp.newBuilder().setSeconds(aLongPoint.getEpochNanos() / NANO_PER_SECOND).setNanos(0).build())
+        .build();
+    Point expectedPoint = Point.newBuilder()
+        .setValue(TypedValue.newBuilder().setInt64Value(((LongPoint) aLongPoint).getValue()))
+        .setInterval(expectedTimeInterval)
+        .build();
+    TimeSeries expectedTimeSeries = TimeSeries.newBuilder()
+        .setMetric(Metric.newBuilder().setType(expectedDescriptor.getType()).putLabels("label1", "value1")
+            .putLabels("label2", "False").build())
+        .addPoints(expectedPoint)
+        .setResource(MonitoredResource.newBuilder().build())
+        .setMetricKind(expectedDescriptor.getMetricKind())
+        .build();
+    CreateMetricDescriptorRequest expectedRequest = CreateMetricDescriptorRequest.newBuilder()
+        .setName("projects/" + aProjectId)
+        .setMetricDescriptor(expectedDescriptor)
+        .build();
+    ProjectName expectedProjectName = ProjectName.of(aProjectId);
+
+    MetricExporter exporter = MetricExporter.createWithClient(aProjectId, mockClient);
+    MetricData metricData = MetricData
+        .create(aMonotonicLongDescriptor, aGceResource, anInstrumentationLibraryInfo, ImmutableList.of(aLongPoint));
+
+    ResultCode result = exporter.export(ImmutableList.of(metricData));
+    verify(mockClient, times(1)).createMetricDescriptor(metricDescriptorCaptor.capture());
+    verify(mockClient, times(1)).createTimeSeries(projectNameArgCaptor.capture(), timeSeriesArgCaptor.capture());
+
+    assertEquals(ResultCode.SUCCESS, result);
+    assertEquals(expectedRequest, metricDescriptorCaptor.getValue());
+    assertEquals(expectedProjectName, projectNameArgCaptor.getValue());
+    assertEquals(1, timeSeriesArgCaptor.getValue().size());
+    assertEquals(expectedTimeSeries, timeSeriesArgCaptor.getValue().get(0));
+  }
+
+  @Test
+  public void testExportWithNonSupportedMetricTypeDoesNothing() {
+    MetricExporter exporter = MetricExporter.createWithClient(aProjectId, mockClient);
+    Descriptor summaryDescriptor = Descriptor
+        .create("Descriptor Name", "Descriptor description", "Unit", Type.SUMMARY,
+            someLabels);
+    MetricData metricData = MetricData
+        .create(summaryDescriptor, aGceResource, anInstrumentationLibraryInfo, ImmutableList.of(aLongPoint));
+
+    ResultCode result = exporter.export(ImmutableList.of(metricData));
+    verify(mockClient, times(0)).createMetricDescriptor(any());
+    verify(mockClient, times(0)).createTimeSeries(any(ProjectName.class), any());
+
+    assertEquals(ResultCode.SUCCESS, result);
```

Comment:
It seems a bit weird to return SUCCESS if all metric data was unsupported and dropped (with a log message)...
I guess it's fine when it was partially successful (only some metrics were dropped).
Could you check if the spec says anything about that?

Reply:
Ah that's a really good point. I can't see anything in the spec, the comments for the `ResultCode` say:
```
enum ResultCode {
    /** The export operation finished successfully. */
    SUCCESS,

    /** The export operation finished with an error. */
    FAILURE
}
```
so I think it does make sense to return FAILURE if some metrics were dropped.